src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -43,6 +43,14 @@ public static class MSBuildLocator
         /// </summary>
         public static bool IsRegistered => s_registeredHandler != null;
 
+        /// <summary>
+        ///     Allow discovery of .NET SDK versions that are unlikely to be successfully loaded in the current process.
+        /// </summary>
+        /// <remarks>
+        ///     Defaults to <see langword="false"/>. Set this to <see langword="true"/> only if your application has special logic to handle loading an incompatible SDK, such as launching a new process with the target SDK's runtime.
+        /// </remarks.
+        public static bool AllowQueryAllRuntimeVersions { get; set; } = false;
```

Comment:
Sorry I only noticed this PR now, but @YuliiaKovalova should a flag like this have been added to the VisualStudioInstanceQueryOptions rather than being a global static? Otherwise it's a bit unclear what a library like Roslyn's MSBuildWorkspace should do here since technically we're just a library and won't ever be registering MSBuild in the calling applications' process at this point -- do we set this and set it back to the old value, and pray there wasn't another thread doing something?

Reply:
Oh, I completely missed that move! I agree with @jasonmalinowski that this made more sense on `VisualStudioInstanceQueryOptions`.